src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -149,6 +151,7 @@ def __str__(self):
         active_directory='https://login.microsoftonline.com',
         active_directory_resource_id='https://management.core.windows.net/',
         active_directory_graph_resource_id='https://graph.windows.net/',
+        active_directory_data_lake_resource_id='https://datalake.azure.net/',
```

Comment:
It's only available on Public Azure?
If so this looks fine but if a user switches cloud and tries to run a command that expected this to be set, the command would fail.
If it's available on other clouds, add this entry there also.